### PR TITLE
Added keys method

### PR DESCRIPTION
**Pull request by wlbrough**
I needed the `keys` method specified in node-cache-manager for a project I am working on

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -136,6 +140,30 @@ class MongooseStore {
...
+        try {
+            return this.model
+                .find({})
+                .then(records => {
+                    if (!records) {
+                        return this.result(fn, 'no records');
+                    }
+
+                    records = records.filter(function(record) {
+                        return (record.exp && record.exp > new Date())
```
Comment (disjunction) [inline src/index.js]: Please move new Date() out of the loop for performance reasons

File: src/index.js [hunk 2]
```diff
@@ -136,6 +140,30 @@ class MongooseStore {
             this.result(fn, e);
         }
     }
+
+    keys(fn) {
+        try {
+            return this.model
+                .find({})
+                .then(records => {
+                    if (!records) {
```
Comment (disjunction) [inline src/index.js]: I don't think this condition is necessary. If there are no records, then an empty array is returned which is treated as true. And if there was an error - we'll run into catch. So i think it's impossible that we get into this condition ever.

File: src/index.js [hunk 3]
```diff
@@ -136,6 +140,30 @@ class MongooseStore {
...
+                        return this.result(fn, 'no records');
+                    }
+
+                    records = records.filter(function(record) {
+                        return (record.exp && record.exp > new Date())
+                    }).map(function(record) {
+                        return record._id;
```
Comment (disjunction) [inline src/index.js]: since we're in ES6 anyway, just map(record => record._id) please

**Comment by wlbrough** [comment]
Made the requested changes. Thanks for the good feedback.

**Comment by disjunction** [comment]
Thanks again. I made a minor change, namely - allow collecting keys also for values stored forever i.e. having no `exp` field.

It's now on npm under version 0.0.4. Cheers!

**Pull request merged**